src/config.c

```diff
@@ -187,10 +188,25 @@ CONFIG_SETTER(setWorkThreads) {
     QueryError_SetErrorFmt(status, QUERY_ELIMIT, "Number of worker threads cannot exceed %d", MAX_WORKER_THREADS);
     return REDISMODULE_ERR;
   }
+
+  int ret = REDISMODULE_OK;
+  size_t res = workersThreadPool_SetNumWorkers(newNumThreads);
+  if (res != newNumThreads) {
+  #ifdef RS_COORDINATOR
+      RedisModule_Log(RSDummyContext, "warning", "Attempt to change the workers thpool size to %lu "
+                                                  "resulted unexpectedly in %lu threads. Updating number of "
+                                                  "connection per shard according to %lu workers.", newNumThreads, res, res);
+  #else
+      RedisModule_Log(RSDummyContext, "warning", "Attempt to change the workers thpool size to %lu "
+                                                  "resulted unexpectedly in %lu threads.", newNumThreads, res);
+  #endif
+    ret = REDISMODULE_ERR;
+  }
```

Comment:
I'm not sure if this is the best way to verify the change.

1. returning `REDISMODULE_ERR` will cause the `FT.CONFIG` command to return an error according to the QueryError values (in this case, `Success (not an error)` because we didn't set it), or cause the module load to fail, all while successfully modifying the configuration value.
2. This is really unexpected. Why not use `RS_LOG_ASSERT` to catch any future weird behavior?

Reply:
yep, assert would be better